backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -784,7 +784,10 @@ private void markCurrentFrameForDeath() {
      */
     public boolean isViewCommand(String command) {
         Vector<Entry> entries = this.getEntriesForCommand(command);
-        return entries.elementAt(0).isView();
+        Entry entry = entries.elementAt(0);
+        return entries.size() == 1
+                && entry.isView()
+                && entry.getPostEntrySessionOperations().size() == 0;
```

Comment:
Add back `entry.getPostEntrySessionOperations().size() == 0` which I should have never taken out at https://github.com/dimagi/commcare/pull/242/files#diff-256562e5e1473541dfae8ff8b3f5bfcbL788